### PR TITLE
feat: Add LogLevel enum with QUIET alias support

### PR DESCRIPTION
**Pull request by CoronRing**
## What does this add?

This PR adds support for `"QUIET"` as a valid string alias for the `logging_setting` parameter, complementing the existing `LogLevel.QUIET` enum. This improves developer experience by matching industry conventions (like `git --quiet`, `pip --quiet`) while maintaining full backward compatibility.

**Problem**: Users expected `logging_setting="QUIET"` to work based on industry conventions, but only `"NONE"` was accepted as a string, causing confusion.

**Solution**: Added `"QUIET"` to the allowable string values, mapping to the same log level as `"NONE"`. Both the enum (`LogLevel.QUIET`) and string (`"QUIET"`) now work seamlessly.

## Type of changes

Please check the type of change your PR introduces:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update (improvements or corrections to documentation)
- [ ] 🎨 Code style/formatting (changes that do not affect the meaning of the code)
- [ ] ♻️ Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] ⚡ Performance improvement (code change that improves performance)
- [x] ✅ Test update (adding missing tests or correcting existing tests)
- [ ] 🔧 Build/CI changes (changes to build process or continuous integration)
- [ ] 🗑️ Chore (other changes that don't modify src or test files)

## Background context

The `LogLevel` enum was introduced to provide IDE autocomplete support and better developer experience. However, only the enum `LogLevel.QUIET` worked, not the string `"QUIET"`. This created an inconsistency where users familiar with Unix tools (`--quiet` flag) couldn't use the string form they expected.

This PR resolves issue #805 by adding string `"QUIET"` support alongside the enum.

### Code Quality
- [x] Code follows the project's style guidelines (run `ruff check .` and `ruff format .`)
- [x] Code is commented, particularly in hard-to-understand areas

### Testing
- [x] Tests added/updated and pass locally (`pytest tests`)
- [x] Test coverage maintained (24 new tests, 51/51 pass, 100% success rate)

### Documentation
- [x] Documentation updated if needed (docstrings enhanced with examples)

### Git & PR Management
- [x] PR title clearly describes the change

### Breaking Changes
- [x] Breaking changes are documented (N/A - no breaking changes)
- [x] Migration guide provided in documentation (N/A - backward compatible)

---

### Usage Examples

All of these methods now work to suppress logs:

```python
import railtracks as rt

# Method 1: Enum QUIET (recommended - IDE autocomplete)
with rt.Session(logging_setting=rt.LogLevel.QUIET):
    result = await rt.call(agent, "Hello")

# Method 2: String "QUIET" (NEW! - matches Unix conventions)
with rt.Session(logging_setting="QUIET"):
    result = await rt.call(agent, "Hello")

# Method 3: Enum NONE (existing)
with rt.Session(logging_setting=rt.LogLevel.NONE):
    result = await rt.call(agent, "Hello")

# Method 4: String "NONE" (existing)
with rt.Session(logging_setting="NONE"):
    result = await rt.call(agent, "Hello")
```

### What Changed

| Component | Before | After |
|-----------|--------|-------|
| String `"QUIET"` | ❌ ValueError | ✅ Accepted |
| Enum `LogLevel.QUIET` | ✅ Works | ✅ Works |
| Backward compatibility | ✅ All strings work | ✅ All strings work |
| Type hints | `Literal["DEBUG", "INFO", ..., "NONE"]` | `Literal["DEBUG", "INFO", ..., "NONE", "QUIET"]` |

### CODE REVIEW

File: packages/railtracks/src/railtracks/utils/logging/config.py [hunk 1]
```diff
@@ -2,17 +2,53 @@
...
+    Attributes:
+        DEBUG: Most verbose logging - shows all internal operations
+        INFO: Standard logging level (default)
+        WARNING: Show warnings and errors only
+        ERROR: Show errors only
+        CRITICAL: Show critical errors only
+        NONE: Suppress all logging output
+
+    Examples:
+        Using LogLevel constants (recommended - enables IDE autocomplete):
```
Comment (soulFood5632) [inline packages/railtracks/src/railtracks/utils/logging/config.py:25]: I would get rid of the (recomonded line becuase lots of tools like vscode have autocomplete for literals as well

File: packages/railtracks/src/railtracks/utils/config.py [hunk 2]
```diff
@@ -46,11 +46,14 @@ def logging_setting(self) -> AllowableLogLevels:
...
+        # Convert LogLevel enum to string value for validation
+        str_value = str(value) if hasattr(value, "value") else value
+
+        if str_value not in str_to_log_level:
             raise ValueError(
-                f"logging_setting must be one of {str_to_log_level}, got {value}"
+                f"logging_setting must be one of {list(str_to_log_level.keys())}, got {value}"
             )
-        self._logging_setting: AllowableLogLevels = value
+        self._logging_setting: AllowableLogLevels = str_value
```
Comment (soulFood5632) [inline packages/railtracks/src/railtracks/utils/config.py:56]: This is a bit of nitpick, but typing wise the `self._logging_setting` can only be a string so we should be typing this properly here

File: packages/railtracks/src/railtracks/_session.py [hunk 3]
```diff
@@ -65,7 +65,18 @@ class Session:
         context (Dict[str, Any], optional): A dictionary of global context variables to be used during the execution.
         timeout (float, optional): The maximum number of seconds to wait for a response to your top-level request.
         end_on_error (bool, optional): If True, the execution will stop when an exception is encountered.
-        logging_setting (AllowableLogLevels, optional): The setting for the level of logging you would like to have. This will override the module-level logging settings for the duration of this session.
+        logging_setting (LogLevel | str, optional): Controls log verbosity. This will override the module-level logging settings for the duration of this session.
+
+            Use LogLevel constants (recommended - enables IDE autocomplete):
```
Comment (soulFood5632) [inline packages/railtracks/src/railtracks/_session.py:70]: same as above, get rid of the line "(recomonded - enables IDE autocomplete)"

File: packages/railtracks/tests/unit_tests/test_loglevel.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+    session1._close()
+    session2._close()
+    session3._close()
+
+
+# ================ END Edge Cases ===============
+
+
+if __name__ == "__main__":
```
Comment (soulFood5632) [inline packages/railtracks/tests/unit_tests/test_loglevel.py:259]: I would delete this line for consitency sake

File: packages/railtracks/tests/unit_tests/test_loglevel.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    docstring = LogLevel.__doc__
+    # Check that all levels are mentioned
+    for level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL", "NONE"]:
+        assert level in docstring
+
+
+# ================ END Documentation Tests ===============
+
+
+# ================= START Integration Tests ===============
```
Comment (soulFood5632) [inline packages/railtracks/tests/unit_tests/test_loglevel.py:195]: I would move this test to either the `Session` unit tests or integration folder

File: packages/railtracks/tests/unit_tests/test_loglevel.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+# ================= START Edge Cases ===============
+
+
+def test_loglevel_none_vs_python_none(mock_session_dependencies):
+    """Test that LogLevel.NONE is different from Python's None"""
+    assert rt.LogLevel.NONE is not None
+    assert str(rt.LogLevel.NONE) == "NONE"
+
+
+def test_session_with_none_logging_uses_global_config(mock_session_dependencies):
```
Comment (soulFood5632) [inline packages/railtracks/tests/unit_tests/test_loglevel.py:224]: same here.
